dot/network/README.md

```diff
@@ -0,0 +1,178 @@
+[key exchange](#identities--key-management), and [Yamux](#yamux) for [stream multiplexing](#stream-multiplexing)), and
+also makes it possible to implement the blockchain-specific protocols defined by Substrate (e.g. [syncing](#sync) and
+[finalising](#GRANDPA) blocks, and maintaining the [transaction pool](#transactions)). The purpose of this document is
+to provide the information that is needed to understand the P2P networking capabilities that are implemented by
+Gossamer - this includes an introduction to P2P networks and `libp2p`, as well as detailed descriptions of the Gossamer
+P2P networking protocols.
+
+## Peer-to-Peer Networking & `libp2p`
+
+[Peer-to-peer](https://en.wikipedia.org/wiki/Peer-to-peer) networking has been a dynamic field of research for over two
+decades, and P2P protocols are at the heart of many blockchain networks. P2P networks can be contrasted with traditional
+[client-server](https://en.wikipedia.org/wiki/Client%E2%80%93server_model) networks where there is a clear separation of
+authority and privilege between the maintainers of the network and its users - in a P2P network, each participant
+possesses equal authority and equal privilege. `libp2p` is a framework for implementing P2P networks that was
+modularized out of [IPFS](https://ipfs.io/); there are implementations in many languages including Go (used by this
+project), Rust, Javascript, C++, and more. In addition to the standard library of protocols in a `libp2p`
+implementation, there is a rich ecosystem of P2P networking packages that work with the pluggable architecture of
+`libp2p`. In some cases, Gossamer uses the `libp2p` networking primitives to implement custom protocols for
+blockchain-specific use cases. What follows is an exploration into three concepts that underpin P2P networks:
+[identity & key management](#identity--key-management), [peer discovery & management](#peer-discovery--management), and
+[stream multiplexing](#stream-multiplexing).
+
+### Identity & Key Management
+
+Many peer-to-peer networks, including those built with Gossamer, use
+[public-key cryptography](https://en.wikipedia.org/wiki/Public-key_cryptography) (also known as asymmetric cryptography)
+to allow network participants to securely identify themselves and interact with one another. The term "asymmetric"
+refers to the fact that in a public-key cryptography system, each participant's identity is associated with a set of two
+keys, each of which serve a distinct ("asymmetric") purpose. One of the keys in an asymmetric key pair is private and is
+used by the network participant to "sign" messages in order to cryptographically prove that the message originated from
+the private key's owner; the other key is public, this is the key that the participant uses to identify themselves - it
+is distributed to network peers to allow for the decryption of messages signed by the corresponding public key. It may
```

Comment:
```suggestion
is distributed to network peers to allow for the verification of messages signed by the corresponding private key. It may
```

this was a bit confusing, it first talks about signing then about decryption?